Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -66,6 +69,9 @@ internal class HttpCoreDiagnosticSourceListener : IObserver<KeyValuePair<string,
             this.client = new TelemetryClient(configuration);
             this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceCore + ":");
 
+            var httpClientVersion = typeof(HttpClient).GetTypeInfo().Assembly.GetName().Version;
+            this.isNetCore20HttpClient = httpClientVersion.CompareTo(new Version(4, 2)) >= 0;
```

Comment:
perhaps you can also remove this code from subscriber and pass isNetCore20HttpClient  to subscriber's constructor

Reply:
right.. less code duplication :)